src/Compiler/Service/ServiceAnalysis.fs

```diff
@@ -451,3 +451,74 @@ module UnusedDeclarations =
             let unusedRanges = getUnusedDeclarationRanges allSymbolUsesInFile isScriptFile
             return unusedRanges
         }
+
+module public Naming =
+
+    type UnnamedFieldsLocations =
+        | InDiscriminatedUnionCases
+        | InExceptions
+
+    let defaultOptions = Set.ofArray [|InDiscriminatedUnionCases; InExceptions|]
+
+    let fieldNotOk (field: FSharpField) =
+        field.IsNameGenerated 
+        || isNull field.Name 
+        || System.String.Empty = field.Name
+
+    [<TailCall>]
```

Comment:
Probably not a good idea until we move to a new released `FSharp.Core` which has this attribute already.

Reply:
Thanks & understood, fresh paint stuff in FSharp.Core needs at least one official release, is it how to look at it?

I used it because I was unsure my code was `TailCall` or `StackOverflow` (due to `yield!`), I don't know how deeply nested code is going to be passed, but I'm hopeful there is headroom in case I missed something making it non TCR-Optimized.

Reply:
> Thanks & understood, fresh paint stuff in FSharp.Core needs at least one official release, is it how to look at it?

Yeah, pretty much. Because some "flavours" of FCS are build with FSharp.Core from nuget. I know @auduchinok uses it.

Reply:
Are there CI empowered people that would put a hard check on this?

I don't know how that would play, beside having a "public surface area" diff and analysis to see which members are getting exposed in the FCS codebase.

Maybe there are other ways?

Just speculating this could be a removed burden for code reviewers (if we fine a robust way of doing).

Reply:
It's already there, but it's not failing CI, since I wasn't sure it was stable enough:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=352682&view=logs&j=119f12ab-97c9-53f0-7ea6-00e6f97fda11&t=5269ef62-afc8-5810-682c-8c493ccd0e61

Reply:
Stellar team!